### PR TITLE
Advances OR-Tools routing app

### PR DESCRIPTION
**Pull request by merschformann**
# Description

Advances the _routing-ortools_ app to incorporate _capacity_ and _max_duration_ constraints while allowing costs driven by duration matrix, distance matrix + speed or haversine + speed.
Furthermore, adapts the app to typical Nextmv routing format to allow plotting in console.

### CODE REVIEW

**Review by nilsbeck — APPROVED** [review]

File: routing-ortools/main.py [hunk 1]
```diff
@@ -32,46 +36,308 @@ def main():
...
+    term1 = np.sin(delta_lat / 2.0) ** 2
+    term2 = np.cos(lats_origin) * np.cos(lats_destination) * np.sin(delta_lon / 2.0) ** 2
+    a = term1 + term2
+    c = 2 * np.arcsin(np.sqrt(a))
+    earth_radius = 6371000
+
+    return earth_radius * c
+
+
 def solve(input_data: dict[str, Any], duration: int) -> dict[str, Any]:
```
Comment (sebastian-quintero) [inline routing-ortools/main.py:55]: I think the `solve` func should come right after `main`, to be consistent with the other apps.

File: routing-ortools/main.py [hunk 2]
```diff
@@ -32,46 +36,308 @@ def main():
...
+                if key not in vehicle:
+                    vehicle[key] = value
+    if "stops" in defaults:
+        for stop in input_data["stops"]:
+            for key, value in defaults["stops"].items():
+                if key not in stop:
+                    stop[key] = value
+
+
+def _valid_location(element: dict[str, Any]) -> bool:
```
Comment (sebastian-quintero) [inline routing-ortools/main.py]: What is the difference here between preceding a function with `_` vs not? I believe convention is to use `_` for private methods that shouldn’t be visible outside the module. For this simple example I would say that methods don’t necessarily need to be preceded by `_`.

**Review by sebastian-quintero — APPROVED** [review]
Style nits

File: routing-ortools/main.py (hunk 2 above)
Comment (merschformann) [reply routing-ortools/main.py]: I agree. This file just grew into considerable complexity, so, this was an attempt to slightly distinguish between auxiliary and more core funcs. However, let me dial that back and re-organize the file a bit (addressing above comment as well).

**Pull request merged**